pkg/apis/pipeline/v1beta1/taskrun_validation_test.go

```diff
@@ -64,6 +64,120 @@ func TestTaskRun_Validate(t *testing.T) {
 			ObjectMeta: metav1.ObjectMeta{Name: "taskrname"},
 		},
 		wc: apis.WithinDelete,
+	}, {
+		name: "propagating params with taskrun",
+		taskRun: &v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{Name: "tr"},
+			Spec: v1beta1.TaskRunSpec{
+				Params: []v1beta1.Param{{
+					Name: "task-words",
+					Value: v1beta1.ArrayOrString{
+						ArrayVal: []string{"hello", "task run"},
+					},
+				}},
+				TaskSpec: &v1beta1.TaskSpec{
+					Steps: []v1beta1.Step{{
+						Name:    "echo",
+						Image:   "ubuntu",
+						Command: []string{"echo"},
+						Args:    []string{"$(params.task-words[*])"},
+					}},
+				},
+			},
+		},
+		wc: config.EnableAlphaAPIFields,
+	}, {
+		name: "propagating partial params with different provided and default names",
+		taskRun: &v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{Name: "tr"},
+			Spec: v1beta1.TaskRunSpec{
+				Params: []v1beta1.Param{{
+					Name: "task-words",
+					Value: v1beta1.ArrayOrString{
+						ArrayVal: []string{"hello", "task run"},
+					},
+				}},
+				TaskSpec: &v1beta1.TaskSpec{
+					Params: []v1beta1.ParamSpec{{
+						Name: "default-task-words",
+						Type: v1beta1.ParamTypeArray,
+					}},
+					Steps: []v1beta1.Step{{
+						Name:    "task-echo",
+						Image:   "ubuntu",
+						Command: []string{"echo"},
+						Args:    []string{"$(params.task-words[*])"},
+					}, {
+						Name:    "default-echo",
+						Image:   "ubuntu",
+						Command: []string{"echo"},
+						Args:    []string{"$(params.default-task-words[*])"},
+					}},
+				},
+			},
+		},
+		wc: config.EnableAlphaAPIFields,
+	}, {
+		name: "propagating partial params in taskrun",
+		taskRun: &v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{Name: "tr"},
+			Spec: v1beta1.TaskRunSpec{
+				Params: []v1beta1.Param{{
+					Name: "task-words",
+					Value: v1beta1.ArrayOrString{
+						ArrayVal: []string{"hello", "task run"},
+					},
+				}},
+				TaskSpec: &v1beta1.TaskSpec{
+					Params: []v1beta1.ParamSpec{{
+						Name: "default-task-words",
+						Type: v1beta1.ParamTypeArray,
+					},
+						{
```

Comment:
could you fix the formatting here please?

Reply:
Yes, of course.

Reply:
Done